lib/ohai/plugins/linux/network.rb

```diff
@@ -406,17 +412,25 @@ def parse_ip_addr_inet6_line(cint, iface, line)
             ]
           end.first
 
-          unless route.nil? or route.empty?
+          if route && !route.empty?
             if family[:name] == "inet"
               ipaddress route[:src]
-              macaddress iface[route[:dev]][:addresses].select{|k,v| v["family"]=="lladdr"}.first.first unless iface[route[:dev]][:flags].include? "NOARP"
+              m = get_mac_for_interface(iface, route[:dev])
+              Ohai::Log.debug("Overwriting macaddress #{macaddress} with #{m} from interface #{route[:dev]}") if macaddress
+              macaddress m
             else
               ip6address route[:src]
+              if macaddress
+                Ohai::Log.debug("Not setting macaddress from ipv6 interface #{route[:dev]} because macaddress is already set")
+              else
+                macaddress get_mac_for_interface(iface, route[:dev])
+              end
             end
+          else
+            Ohai::Log.debug("Unable to determine ideal default route, not setting ipaddress/ip6address/macaddress. Should have considered #{default_route[:dev]}?")
```

Comment:
I may be mis-understanding the code... but you end up doing the exact same thing in `plugins/network.rb` anyway, so why not do it here? Also, it's one thing to _change_ the behavior for an existing case, but I can't imagine it's *ever* OK to just ... NOT pick a macaddress. That seems just broken in any and all use-cases.